### PR TITLE
raptor: fix the duration of thz section after a stayin

### PR DESCRIPTION
**Pull request by kinnou02**
If the arrival and the departure was not the same the duration was
negative

### CODE REVIEW

File: source/jormungandr/tests/routing_tests.py [hunk 1]
```diff
@@ -280,7 +281,7 @@ def test_max_attemps(self):
             .format(from_sa="E", to_sa="H", datetime="20120615T080000")
 
         response = self.query_region(query, display=False)
-        assert(not "journeys" in response or len(response['journeys']) == 0)
+        assert (not "journeys" in response or len(response['journeys']) ==  0)
```
Comment (TeXitoi) [inline source/jormungandr/tests/routing_tests.py]: ???

File: source/routing/raptor_api.cpp [hunk 2]
```diff
@@ -235,25 +235,29 @@ pbnavitia::Response make_pathes(const std::vector<navitia::routing::Path>& paths
...
-
                 bt::time_period action_period(item.departure, item.arrival);
                 const auto origin_sp = item.stop_points.front();
                 const auto destination_sp = item.stop_points.back();
-                fill_pb_placemark(origin_sp, d, pb_section->mutable_origin(), 1,
-                        now, action_period, show_codes);
-                fill_pb_placemark(destination_sp, d, pb_section->mutable_destination(), 1,
-                        now, action_period, show_codes);
+                fill_pb_placemark(origin_sp, d, pb_section->mutable_origin(), 1, now, action_period, show_codes);
+                fill_pb_placemark(destination_sp, d, pb_section->mutable_destination(), 1, now, action_period, show_codes);
```
Comment (TeXitoi) [inline source/routing/raptor_api.cpp]: long line

File: source/routing/raptor_api.cpp [hunk 3]
```diff
@@ -235,25 +235,29 @@ pbnavitia::Response make_pathes(const std::vector<navitia::routing::Path>& paths
...
                 pb_section->set_length(origin_sp->coord.distance_to(destination_sp->coord));
             }
-            auto dep_time = navitia::to_posix_timestamp(item.departure);
+            uint64_t dep_time, arr_time;
+            if(item.stop_points.size() == 1 && item.type == public_transport){
+                //after a stay it's possible to have only one point, so we take the arrival hour as arrival and departure
```
Comment (TeXitoi) [inline source/routing/raptor_api.cpp]: long line

File: source/routing/raptor_api.cpp [hunk 4]
```diff
@@ -235,25 +235,29 @@ pbnavitia::Response make_pathes(const std::vector<navitia::routing::Path>& paths
...
+            }
             pb_section->set_begin_date_time(dep_time);
-            auto arr_time = navitia::to_posix_timestamp(item.arrival);
             pb_section->set_end_date_time(arr_time);
 
             if(departure_time == bt::pos_infin)
                 departure_time = item.departure;
             arrival_time = item.arrival;
-            pb_section->set_duration((item.arrival - item.departure).total_seconds());
+            pb_section->set_duration(pt::seconds(arr_time - dep_time).total_seconds());
```
Comment (TeXitoi) [inline source/routing/raptor_api.cpp]: convertion do uint32_t -> pt::duration -> uint32_t???

File: source/routing/tests/routing_api_test.cpp [hunk 5]
```diff
@@ -431,7 +431,7 @@ BOOST_AUTO_TEST_CASE(journey_arrival_in_a_stay_in) {
                                             ("btl",  61560,61560)
                                             ("vg",   61920,61920)
                                             ("ht:4", 62340,62340);
-    b.vj("4462", "1111111", "block1", true) ("end",62400,62400)
+    b.vj("4462", "1111111", "block1", true) ("end", 62400,62500)
```
Comment (antoine-de) [inline source/routing/tests/routing_api_test.cpp]: you need an extra space

**Pull request merged**